# Dublin Airbnb Market Analysis
## Notebook 01 — Data Collection & Inspection

**Source:** Inside Airbnb — insideairbnb.com  
**Scrape date:** September 16, 2025  
**Geography:** Dublin, Ireland  

Just loading the raw files and seeing what we're working with. No cleaning here — that's notebook 02.

In [12]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import os
pd.set_option('display.max_columns', 100)

In [13]:
# change this to wherever you put the files in your drive
base_path = '/content/drive/MyDrive/Colab Notebooks/Airbnb/'

listings_path  = '/content/listings.csv.gz'
calendar_path  = '/content/calendar.csv.gz'
reviews_path   = '/content/reviews.csv.gz'
geo_path       = '/content/neighbourhoods.geojson'

---
## 1. Listings

In [14]:
# loading listings first as its the main file
listings = pd.read_csv(listings_path, compression='gzip')

print("Shape:", listings.shape)
print("\nColumns:\n", listings.columns.tolist())

Shape: (6945, 79)

Columns:
 ['id', 'listing_url', 'scrape_id', 'last_scraped', 'source', 'name', 'description', 'neighborhood_overview', 'picture_url', 'host_id', 'host_url', 'host_name', 'host_since', 'host_location', 'host_about', 'host_response_time', 'host_response_rate', 'host_acceptance_rate', 'host_is_superhost', 'host_thumbnail_url', 'host_picture_url', 'host_neighbourhood', 'host_listings_count', 'host_total_listings_count', 'host_verifications', 'host_has_profile_pic', 'host_identity_verified', 'neighbourhood', 'neighbourhood_cleansed', 'neighbourhood_group_cleansed', 'latitude', 'longitude', 'property_type', 'room_type', 'accommodates', 'bathrooms', 'bathrooms_text', 'bedrooms', 'beds', 'amenities', 'price', 'minimum_nights', 'maximum_nights', 'minimum_minimum_nights', 'maximum_minimum_nights', 'minimum_maximum_nights', 'maximum_maximum_nights', 'minimum_nights_avg_ntm', 'maximum_nights_avg_ntm', 'calendar_updated', 'has_availability', 'availability_30', 'availability_60', 

6,945 listings, 79 columns. Lot of columns — plenty of them are URLs and free text we won't need.

In [15]:
listings.head()

,id,listing_url,scrape_id,last_scraped,source,name,description,neighborhood_overview,picture_url,host_id,host_url,host_name,host_since,host_location,host_about,host_response_time,host_response_rate,host_acceptance_rate,host_is_superhost,host_thumbnail_url,host_picture_url,host_neighbourhood,host_listings_count,host_total_listings_count,host_verifications,host_has_profile_pic,host_identity_verified,neighbourhood,neighbourhood_cleansed,neighbourhood_group_cleansed,latitude,longitude,property_type,room_type,accommodates,bathrooms,bathrooms_text,bedrooms,beds,amenities,price,minimum_nights,maximum_nights,minimum_minimum_nights,maximum_minimum_nights,minimum_maximum_nights,maximum_maximum_nights,minimum_nights_avg_ntm,maximum_nights_avg_ntm,calendar_updated,has_availability,availability_30,availability_60,availability_90,availability_365,calendar_last_scraped,number_of_reviews,number_of_reviews_ltm,number_of_reviews_l30d,availability_eoy,number_of_reviews_ly,estimated_occupancy_l365d,estimated_revenue_l365d,first_review,last_review,review_scores_rating,review_scores_accuracy,review_scores_cleanliness,review_scores_checkin,review_scores_communication,review_scores_location,review_scores_value,license,instant_bookable,calculated_host_listings_count,calculated_host_listings_count_entire_homes,calculated_host_listings_count_private_rooms,calculated_host_listings_count_shared_rooms,reviews_per_month
0,44077,https://www.airbnb.com/rooms/44077,20250916040708,2025-09-16,city scrape,cosy comfortable Irish cottage twin,Our house is a cosy comfortable cottage with ...,I like our neighbourhood as there is no shorta...,https://a0.muscache.com/pictures/525706/050a3a...,193005,https://www.airbnb.com/users/show/193005,Teresa,2010-08-06,"Dublin, Ireland","I am a friendly outgoing Irishwoman ,who love...",within an hour,100%,97%,t,https://a0.muscache.com/im/pictures/user/2167f...,https://a0.muscache.com/im/pictures/user/2167f...,Churchtown,2.0,2.0,"['email', 'phone']",t,t,"Churchtown, Ireland",Dn Laoghaire-Rathdown,NaN,53.29178,-6.25792,Private room in bungalow,Private room,2,1.0,1 private bath,1.0,2.0,"[""Paid washer \u2013 In unit"", ""Wine glasses"",...",$86.00,3,30,3.0,3.0,30.0,30.0,3.0,30.0,NaN,t,6,9,9,9,2025-09-16,394,37,5,9,42,222,19092.0,2011-03-20,2025-09-13,4.82,4.86,4.77,4.93,4.93,4.68,4.83,NaN,f,2,0,2,0,2.23
1,85156,https://www.airbnb.com/rooms/85156,20250916040708,2025-09-16,previous scrape,Cosy Comfortable Irish Cottage 1 Double Bed,Our Cottage is a charming light filled cottage...,"I love Dundrum and its surrounding areas ,<br ...",https://a0.muscache.com/pictures/1749253/9ed2a...,193005,https://www.airbnb.com/users/show/193005,Teresa,2010-08-06,"Dublin, Ireland","I am a friendly outgoing Irishwoman ,who love...",within an hour,100%,97%,t,https://a0.muscache.com/im/pictures/user/2167f...,https://a0.muscache.com/im/pictures/user/2167f...,Churchtown,2.0,2.0,"['email', 'phone']",t,t,"Churchtown, Ireland",Dn Laoghaire-Rathdown,NaN,53.29209,-6.25624,Private room in bungalow,Private room,2,NaN,1.5 baths,NaN,NaN,"[""Wine glasses"", ""Coffee maker"", ""Coffee"", ""Sh...",NaN,3,21,3.0,3.0,21.0,21.0,3.0,21.0,NaN,t,7,7,7,7,2025-09-16,357,41,5,7,41,246,NaN,2011-05-24,2025-09-14,4.81,4.84,4.75,4.91,4.89,4.64,4.77,NaN,f,2,0,2,0,2.05
2,159889,https://www.airbnb.com/rooms/159889,20250916040708,2025-09-16,city scrape,Friendly Single Room,Washing can be done at a cost of €5 per load....,Plenty of buses into the city and the area is ...,https://a0.muscache.com/pictures/3031697/a8259...,766611,https://www.airbnb.com/users/show/766611,Marie,2011-07-01,"Dublin Airport, Ireland",Used to host students and now I am looking fo...,within a few hours,70%,91%,f,https://a0.muscache.com/im/pictures/user/5059e...,https://a0.muscache.com/im/pictures/user/5059e...,Beaumont,3.0,3.0,"['email', 'phone']",t,t,"Dublin, Ireland",Dublin City,NaN,53.39027,-6.23547,Private room in home,Private room,1,2.0,2 shared baths,1.0,1.0,"[""Private patio or balcony"", ""Paid washer"", "

- `price` is `'$86.00'` — string with dollar sign, needs converting
- `host_is_superhost` is `'t'`/`'f'` not boolean
- `neighbourhood_group_cleansed` looks completely empty
- some listings have NaN price already (rows 1, 4)

In [16]:
# checking data types and how many nulls we have
listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6945 entries, 0 to 6944
Data columns (total 79 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            6945 non-null   int64  
 1   listing_url                                   6945 non-null   object 
 2   scrape_id                                     6945 non-null   int64  
 3   last_scraped                                  6945 non-null   object 
 4   source                                        6945 non-null   object 
 5   name                                          6945 non-null   object 
 6   description                                   6827 non-null   object 
 7   neighborhood_overview                         2480 non-null   object 
 8   picture_url                                   6945 non-null   object 
 9   host_id                                       6945 non-null   i

3 columns are entirely null — `license`, `calendar_updated`, `neighbourhood_group_cleansed`. Dropping those.

`price` has 1,701 nulls — about 24% of listings. Can't do anything with those so they're getting dropped.

Review score nulls (1,042) are all the same listings — ones with zero reviews. That's fine, just means they're new.

In [17]:
# need to check what format price is in before we do anything with it
print(listings['price'].dtype)
print("\nSample values:")
print(listings['price'].head(10))

object

Sample values:
0     $86.00
1        NaN
2     $48.00
3    $101.00
4        NaN
5     $30.00
6        NaN
7    $130.00
8    $177.00
9     $87.00
Name: price, dtype: object


Confirmed — `'$86.00'` format. Dollar sign even though it's Dublin data. Values are in euros, just stored with `$`.

In [18]:
# how many missing values in each column
null_counts = listings.isnull().sum()
null_counts = null_counts[null_counts > 0].sort_values(ascending=False)

print("Columns with missing values:\n")
print(null_counts)

Columns with missing values:

license                         6945
calendar_updated                6945
neighbourhood_group_cleansed    6945
neighbourhood                   4465
neighborhood_overview           4465
host_about                      4153
host_location                   1802
bathrooms                       1708
beds                            1704
estimated_revenue_l365d         1701
price                           1701
host_neighbourhood              1410
review_scores_cleanliness       1042
review_scores_value             1042
review_scores_rating            1042
first_review                    1042
review_scores_accuracy          1042
review_scores_location          1042
review_scores_communication     1042
reviews_per_month               1042
last_review                     1042
review_scores_checkin           1042
host_response_time               914
host_response_rate               914
bedrooms                         631
host_acceptance_rate             460
host_is_

`license` is 100% null across all 6,945 rows — confirms what we noted in the scope, no compliance data available in this dataset.

Plan for cleaning:
- drop the 3 empty columns + URL/text cols
- drop rows with null price
- fill `bedrooms`, `beds`, `bathrooms` nulls with median
- leave host rate nulls as-is, don't want to fake that

---
## 2. Reviews

In [19]:
reviews = pd.read_csv(reviews_path, compression='gzip')

print("Shape:", reviews.shape)
print("\nFirst few rows:")
reviews.head()

Shape: (334122, 6)

First few rows:


,listing_id,id,date,reviewer_id,reviewer_name,comments
0,44077,203703,2011-03-20,393348,Christian,We enjoyed our stay very much. The room was co...
1,44077,211369,2011-03-28,444004,Solidea,We have been here 4 nights. Stay in a home is ...
2,44077,234215,2011-04-21,465058,Michael And Isabelle,Teresa and Hughie were great hosts. They were ...
3,44077,261843,2011-05-13,490005,Weston,"No surprises, was as described. Very gracious..."
4,44077,268148,2011-05-17,520460,Barbara,"Teresa was a lovely hostess, and we had a deli..."


334,122 reviews, 6 columns. Clean and simple — `listing_id`, `date`, `reviewer_id`, `reviewer_name`, `comments`.

Earliest review in the sample is 2011 — long history. Using this mainly for review volume and recency, not the text.

---
## 3. Calendar

In [20]:
# calendar is huge so just loading first 10000 rows to inspect
# we'll load the full thing properly in the cleaning notebook
calendar_sample = pd.read_csv(calendar_path, compression='gzip', nrows=10000)

print("Shape of sample:", calendar_sample.shape)
print("\nColumns:", calendar_sample.columns.tolist())
print("\nSample rows:")
calendar_sample.head()

Shape of sample: (10000, 7)

Columns: ['listing_id', 'date', 'available', 'price', 'adjusted_price', 'minimum_nights', 'maximum_nights']

Sample rows:


,listing_id,date,available,price,adjusted_price,minimum_nights,maximum_nights
0,10146203,2025-09-16,f,NaN,NaN,2,55
1,10146203,2025-09-17,f,NaN,NaN,2,55
2,10146203,2025-09-18,f,NaN,NaN,2,55
3,10146203,2025-09-19,f,NaN,NaN,2,55
4,10146203,2025-09-20,t,NaN,NaN,2,55


`price` and `adjusted_price` are both fully null — calendar pricing is empty, we use listings price instead.

`available` column is fine though — that's what we need for occupancy and seasonality analysis.

---
## 4. GeoJSON

In [21]:
import json

with open(geo_path, 'r') as f:
    geo = json.load(f)

print("Type:", geo['type'])
print("Number of neighbourhoods:", len(geo['features']))
print("\nAll neighbourhood names:")
for feature in geo['features']:
    print(" -", feature['properties'])

Type: FeatureCollection
Number of neighbourhoods: 4

All neighbourhood names:
 - {'neighbourhood': 'Dublin City', 'neighbourhood_group': None}
 - {'neighbourhood': 'South Dublin', 'neighbourhood_group': None}
 - {'neighbourhood': 'Fingal', 'neighbourhood_group': None}
 - {'neighbourhood': 'Dn Laoghaire-Rathdown', 'neighbourhood_group': None}


Only 4 boundaries — Dublin City, Dún Laoghaire-Rathdown, Fingal, South Dublin.

These are the 4 local authority areas, not granular neighbourhoods. Choropleth maps will be at this level.

In [22]:
# neighbourhood_cleansed is our main geography column — check how many areas we have
print("Unique values in neighbourhood_cleansed:")
print(listings['neighbourhood_cleansed'].value_counts())

Unique values in neighbourhood_cleansed:
neighbourhood_cleansed
Dublin City              4815
Fingal                    853
Dn Laoghaire-Rathdown     792
South Dublin              485
Name: count, dtype: int64


Matches the GeoJSON exactly — 4 areas. Dublin City has the bulk of listings as expected.

In [23]:
# double checking the actual scrape date from the data itself
print("Scrape dates in dataset:")
print(listings['last_scraped'].value_counts())

Scrape dates in dataset:
last_scraped
2025-09-16    6945
Name: count, dtype: int64


All 6,945 listings scraped on the same date — clean single snapshot, no mixing of old and new scrapes.

---
## What needs fixing in Notebook 02

- strip `$` from price → convert to float
- drop empty columns: `license`, `calendar_updated`, `neighbourhood_group_cleansed`
- drop URL, image, free text columns
- drop rows with null price
- remove outlier prices (€0 and above €1,500)
- convert `t`/`f` to boolean: `host_is_superhost`, `instant_bookable`, `has_availability`
- convert percentage strings to numeric: `host_response_rate`, `host_acceptance_rate`
- fill nulls: `bedrooms`, `beds`, `bathrooms` → median
- engineer `occupancy_rate` and `estimated_annual_revenue`